rafay/gke_flatten_util.go

```diff
@@ -0,0 +1,744 @@
+package rafay
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/RafaySystems/rafay-common/proto/types/hub/infrapb"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// convert from V3 schema in rafay-common proto --> tf schema
+func flattenGKEClusterV3(d *schema.ResourceData, in *infrapb.Cluster) error {
+	/*
+		Cluster:
+		- apiversion
+		- kind
+		- metadata
+		- spec
+	*/
+	log.Println("flattenGKEClusterV3 in", in)
+	if in == nil {
+		return nil
+	}
+	obj := map[string]interface{}{}
+
+	if len(in.ApiVersion) > 0 {
+		obj["api_version"] = in.ApiVersion
+	}
+	if len(in.Kind) > 0 {
+		obj["kind"] = in.Kind
+	}
+	var err error
+
+	var ret1 []interface{}
+	if in.Metadata != nil {
+		v, ok := obj["metadata"].([]interface{})
+		if !ok {
+			v = []interface{}{}
+		}
+		ret1 = flattenMetadataV3(in.Metadata, v)
```

Comment:
@poornima-rafay Out of curiousity..why can't we use flattenMetada function itself?

Reply:
Suggestion: we need not fetch 'v' and pass it to this function. It's anyway not being used.

Reply:
@MJjainam good question. I am not sure either. I am reusing the function that was written for AKS_V3. 
Also I see there's some difference between flattenMetaData() and flattenMetadataV3(). 
@thirumal-rafay do you know why we introduced flattenMetadataV3 for AKS?